### ISSUE TITLE
Dockerfile: Volta currently only provides pre-built binaries for x86_64 architectures

### ISSUE THREAD

**Issue by javierav** [rails/rails#47407]
### Steps to reproduce

Given a Rails project with this `Dockerfile` (slightly modified from [here](https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt))

```Dockerfile
# syntax = docker/dockerfile:1

# Make sure it matches the Ruby version in .ruby-version and Gemfile
ARG RUBY_VERSION=3.1.1
FROM ruby:$RUBY_VERSION-slim as base

# Rails app lives here
WORKDIR /app

# Set production environment
ENV RAILS_ENV="production" \
    BUNDLE_WITHOUT="development"


# Throw-away build stage to reduce size of final image
FROM base as build

# Install packages need to build gems and node modules
RUN apt-get update -qq && \
    apt-get install -y build-essential git pkg-config libpq-dev redis libvips curl node-gyp python-is-python3

# Install JavaScript dependencies
ARG NODE_VERSION=18.13.0
ARG YARN_VERSION=1.22.19
ENV VOLTA_HOME="/usr/local"
RUN curl https://get.volta.sh | bash && \
    volta install node@$NODE_VERSION yarn@$YARN_VERSION

# Install application gems
COPY Gemfile Gemfile.lock ./
RUN bundle install && \
    bundle exec bootsnap precompile --gemfile

# Install node modules
COPY package.json yarn.lock ./
RUN yarn install

# Copy application code
COPY . .

# Precompile bootsnap code for faster boot times
RUN bundle exec bootsnap precompile app/ lib/

# Precompiling assets for production without requiring secret RAILS_MASTER_KEY
RUN SECRET_KEY_BASE_DUMMY=1 ./bin/rails assets:precompile


# Final stage for app image
FROM base

RUN apt-get update -qq && \
    apt-get install --no-install-recommends -y postgresql-client redis libvips && \
    rm -rf /var/lib/apt/lists /var/cache/apt/archives

# add custom user
RUN useradd rails

USER rails:rails

# Copy built artifacts: gems, application
COPY --from=build --chown=rails:rails /usr/local/bundle /usr/local/bundle
COPY --from=build --chown=rails:rails /app /app

# Entrypoint prepares the database.
ENTRYPOINT ["/app/bin/docker-entrypoint"]

# Start the server by default, this can be overwritten at runtime
EXPOSE 3000
CMD ["./bin/rails", "server"]
```

Run docker build command:

```
$ docker build .
```

### Expected behavior
Docker build image for ARM64 (Apple Silicon)

### Actual behavior
 => ERROR [build 2/9] RUN curl https://get.volta.sh | bash &&     volta install node@18.13.0 yarn@1.22.19

Error: Sorry! Volta currently only provides pre-built binaries for x86_64 architectures.

### System configuration
**Rails version**: 7.1.0.alpha from git

**Ruby version**: 3.2.0 (2022-12-25 revision a528908271) [arm64-darwin21]

**Comment by yahonda** [comment]
#47441 has been opened.

**Comment by skipkayhil** [comment]
Fixed in #47459
